### PR TITLE
doc: extensions: zephyr-domain: some fixes

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

File: doc/_extensions/zephyr/domain.py [hunk 1]
```diff
@@ -80,6 +80,9 @@ class ConvertCodeSampleNode(SphinxTransform):
     default_priority = 100
 
     def apply(self):
+        if not self.config.zephyr_domain_apply_transforms:
```
Comment (kartben) [inline doc/_extensions/zephyr/domain.py:83]: I'm sure you did it this way for a reason, but wouldn't it be simpler/clearer to simply not install the transforms in the `setup()`? Also see my general comment/question to understand better what this is trying to accomplish

**Review by kartben — COMMENTED** [review]
Could you please clarify if only the first commit was needed to fix your issue of not being able to use the domain extension in the context of Intersphinx?

My concern with the second one, is that you might be hiding useful information now  -- but this is probably due to my own lack of understanding of Intersphinx and the context in which your using it? Like, the code samples page now likely don't have a title anymore and don't get mounted in "your" Zephyr documentation anymore?

In any case, rather than having a `zephyr_domain_apply_transforms` flag I think it might make sense to make it more about the actual config option(s) that you'd want to have access too, ex. `zephyr_code_sample_show_in_doxygen`, ...

Thanks for the PR, and investing time in looking into this

File: doc/_extensions/zephyr/domain.py (hunk 1 above)
Comment (gmarull) [reply doc/_extensions/zephyr/domain.py:83]: I'm not sure how to do that from setup() (Sphinx settings are defined there)

**Comment by gmarull** [comment]
> Could you please clarify if only the first commit was needed to fix your issue of not being able to use the domain extension in the context of Intersphinx?

Only the first is needed. However, I run into weird errors due to transforms, and I realized they're not needed if only using the domain to access references (ie using code-sample role)

> 
> My concern with the second one, is that you might be hiding useful information now -- but this is probably due to my own lack of understanding of Intersphinx and the context in which your using it? Like, the code samples page now likely don't have a title anymore and don't get mounted in "your" Zephyr documentation anymore?
> 
> In any case, rather than having a `zephyr_domain_apply_transforms` flag I think it might make sense to make it more about the actual config option(s) that you'd want to have access too, ex. `zephyr_code_sample_show_in_doxygen`, ...
> 
> Thanks for the PR, and investing time in looking into this

File: doc/_extensions/zephyr/domain.py (hunk 1 above)
Comment (kartben) [reply doc/_extensions/zephyr/domain.py:83]: I mean just don't add them to app at lines L317-L321?

File: doc/conf.py [hunk 2]
```diff
@@ -310,6 +310,10 @@
     "build/dts/api/compatibles/**/*",
 ]
 
+# -- Options for zephyr.domain --------------------------------------------
+
+zephyr_domain_apply_transforms = True
```
Comment (kartben) [inline doc/conf.py:315]: until we settle on whether it's needed at all or needs to be tweaked, and before i forget, I would drop "_domain" from to be consistent with existing naming scheme of other Sphinx domains https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-c-domain 

```suggestion
zephyr_apply_transforms = True
```

**Comment by gmarull** [comment]
@kartben please re-review, I've reworked the transforms part to only affect breathe.

**Review by kartben — APPROVED** [review]
Perfect, thank you @gmarull!
FWIW I still think it would be more natural to not install the monkey patch in the first place (L327) when the config flag is false, but your solution certainly works too :)

**Comment by gmarull** [comment]
> Perfect, thank you @gmarull! FWIW I still think it would be more natural to not install the monkey patch in the first place (L327) when the config flag is false, but your solution certainly works too :)

I agree, but I'm not sure how to do that programmatically from conf.py...

**Comment by carlescufi** [comment]
Labelling this as Hotfix because as it is it breaks integration of the Zephyr doc with other Sphinx docsets.

**Review by carlescufi — APPROVED** [review]

**Comment by kartben** [comment]
> Labelling this as Hotfix because as it is it breaks integration of the Zephyr doc with other Sphinx docsets.

FWIW I am not certain this qualifies as something that's "blocking developers from doing their daily work", given that this is something that's been there for over 3 months. Is this a serious bug for downstream adopters, surely, but is it blocking upstream developers from doing their work, probably not.
All this to say that it would be great to not fast track this one if possible :)

**Comment by kartben** [comment]
> > Perfect, thank you @gmarull! FWIW I still think it would be more natural to not install the monkey patch in the first place (L327) when the config flag is false, but your solution certainly works too :)
> 
> I agree, but I'm not sure how to do that programmatically from conf.py...

Damn, yeah, I finally see what you mean. And yeah, I guess that's just how we have to deal with Sphinx's configuration framework.

**Comment by carlescufi** [comment]
> All this to say that it would be great to not fast track this one if possible :)

Alright, let's wait then.

**Review by tejlmand — APPROVED** [review]

**Comment by kartben** [comment]
> > All this to say that it would be great to not fast track this one if possible :)
> 
> Alright, let's wait then.

Thanks!

BTW @gmarull -- maybe doc infrastructure needs to have some level of integration tests to ensure that downstream users relying on documentation "features"/API don't get regressions. I don't know how easy it is to have a minimal intersphinx setup but maybe this could be a starting point, i.e making sure that a basic downstream docset can properly make use of our various domains, etc.?

**Pull request merged**